includes/polling/ports.inc.php

```diff
@@ -363,6 +363,54 @@
     }
 }
 
+//NAC Polling
+if ($config['enable_ports_nac']) {
+    if ($device['os'] == 'iosxe' || 'ios') {
+        echo "\nCisco-NAC\n";
+        $PortAuthSessionEntry = snmpwalk_cache_oid($device, 'cafSessionEntry', array(), 'CISCO-AUTH-FRAMEWORK-MIB');
+        foreach ($PortAuthSessionEntry as $PortAuthSessionEntryNAC => $PortAuthSessionEntryParameters) {
+            $port_index_nac = substr($PortAuthSessionEntryNAC, 0, strpos($PortAuthSessionEntryNAC, "."));
+            $port_auth_id_nac = strstr($PortAuthSessionEntryNAC, "'" );
+            $port_auth_id_nac = substr($port_auth_id_nac, 0, -1 );
+            $port_auth_id_nac = substr($port_auth_id_nac, 1);
+            $dbreult = dbFetchRow('SELECT * FROM `ports_nac` WHERE `auth_id` = ?' , $port_auth_id_nac);
+            if ($dbreult == null){
+                echo "Not found\n";
+                dbInsert(array('auth_id' => $port_auth_id_nac), 'ports_nac');
+            }
+            $IPHextpDec = explode(' ',$PortAuthSessionEntryParameters['cafSessionClientAddress']);
```

Comment:
Try IP::fromHextString() here

Reply:
`use LibreNMS\Util\IP;`

Reply:
You will also need to surround it with a try/catch block in case it gets invalid input.  Putting all the code that relies on it returning a correct IP inside the try/catch.